### PR TITLE
[ECP-8309] - Support Tokenization on In Person payments. 

### PR DESCRIPTION
**Pull request by raoulritter**
**Description**

Simplify tokenization for In Person payments and add it to the admin configuration and settings in the same manner as other payment methods. 

**Tested scenarios**

Fixes

### CODE REVIEW

File: Gateway/Response/PaymentPosCloudHandler.php [hunk 1]
```diff
@@ -43,41 +45,15 @@ public function handle(array $handlingSubject, array $response)
         // do not send order confirmation mail
         $payment->getOrder()->setCanSendNewEmailFlag(false);
 
+
         if (!empty($paymentResponse['Response']['AdditionalResponse'])
         ) {
```
Comment (candemiralp) [inline Gateway/Response/PaymentPosCloudHandler.php:50]:
```suggestion
        if (!empty($paymentResponse['Response']['AdditionalResponse'])) {
```

File: Helper/PointOfSale.php [hunk 2]
```diff
@@ -43,12 +45,13 @@ public function addSaleToAcquirerData(array $request, Order $order) : array
 
         // If customer exists add it into the request to store request
         if (!empty($customerId)) {
-            $recurringContract = $this->configHelper->getAdyenPosCloudConfigData('recurring_type', $storeId);
+            $posRecurringEnabled = $this->configHelper->getPaymentMethodRecurringActive(AdyenPosCloudConfigProvider::CODE ,$storeId);
```
Comment (candemiralp) [inline Helper/PointOfSale.php:48]:
```suggestion
            $posRecurringEnabled = $this->configHelper->getPaymentMethodRecurringActive(AdyenPosCloudConfigProvider::CODE, $storeId);
```

File: Helper/PointOfSale.php [hunk 3]
```diff
@@ -43,12 +45,13 @@ public function addSaleToAcquirerData(array $request, Order $order) : array
...
+            $recurringProcessingModel = $this->configHelper->getPaymentMethodRecurringProcessingModel(AdyenPosCloudConfigProvider::CODE ,$storeId);
```
Comment (candemiralp) [inline Helper/PointOfSale.php:49]:
```suggestion
            $recurringProcessingModel = $this->configHelper->getPaymentMethodRecurringProcessingModel(AdyenPosCloudConfigProvider::CODE, $storeId);
```

File: Model/Config/Backend/Tokenization.php [hunk 4]
```diff
@@ -153,6 +155,16 @@ private function buildPaymentMethodArray(): array
                 $arrayKey = $this->mathRandom->getUniqueHash('_');
                 $result[$arrayKey] = $paymentMethodValue;
             }
+            elseif ($adyenPaymentMethod === AdyenPosCloudConfigProvider::CODE ){
```
Comment (candemiralp) [inline Model/Config/Backend/Tokenization.php:158]:
```suggestion
            } elseif ($adyenPaymentMethod === AdyenPosCloudConfigProvider::CODE ){
```

File: Model/Config/Source/TokenizedPaymentMethods.php
Comment (candemiralp) [inline Model/Config/Source/TokenizedPaymentMethods.php:1]: Could you please remove this file?

Comment (raoulritter) [reply Model/Config/Source/TokenizedPaymentMethods.php:1]: This is the removal of the file.

File: Helper/PointOfSale.php [hunk 5]
```diff
@@ -21,12 +23,12 @@ class PointOfSale
 {
     private Data $dataHelper;
     private ProductMetadataInterface $productMetadata;
-    protected Config $configHelper;
+    private Vault $configHelper;
```
Comment (hossam-adyen) [inline Helper/PointOfSale.php:26]:
```suggestion
    private Vault $vaultHelper;
```

**Review by hossam-adyen — DISMISSED** [review]

File: Model/Config/Backend/Tokenization.php [hunk 6]
```diff
@@ -150,6 +152,15 @@ private function buildPaymentMethodArray(): array
                     'recurring_processing_model' => null
                 ];
 
+                $arrayKey = $this->mathRandom->getUniqueHash('_');
+                $result[$arrayKey] = $paymentMethodValue;
+            } elseif ($adyenPaymentMethod === AdyenPosCloudConfigProvider::CODE ){
```
Comment (candemiralp) [inline Model/Config/Backend/Tokenization.php:157]: With the latest changes in `rc-v9`, this block will create a confliction. Could you please check the latest implementation of this block [here](https://github.com/Adyen/adyen-magento2/blob/31ff703486c930140efdf5503ff86868622e7556/Model/Config/Backend/Tokenization.php#L128) and modify the code? After implementing the new block, you need to specify `<supports_recurring>1</supports_recurring>` in payment methods config block in [config.xml](https://github.com/Adyen/adyen-magento2/blob/31ff703486c930140efdf5503ff86868622e7556/etc/config.xml#L378).

**Review by hossam-adyen — COMMENTED** [review]
Nice work, would love to see some tests to cover the new code lines and files

File: Plugin/FilterSortPaymentMethods.php [hunk 7]
```diff
@@ -54,7 +54,6 @@ public function afterGetAvailableMethods(
         $this->fetchPaymentMethods($quote);
 
         if (!empty($this->fetchPaymentMethodsResponse)) {
-            $this->filterPaymentMethods();
```
Comment (candemiralp) [inline Plugin/FilterSortPaymentMethods.php:57]: This was just for testing. Could you please revert?

File: Model/Config/Source/TokenizedPaymentMethods.php
Comment (candemiralp) [inline Model/Config/Source/TokenizedPaymentMethods.php:1]: Could you please remove this empty file?

File: Plugin/FilterSortPaymentMethods.php (hunk 7 above)
Comment (candemiralp) [inline Plugin/FilterSortPaymentMethods.php:57]: Can you please revert this?

**Review by candemiralp — APPROVED** [review]

**Pull request merged**